superset/db_engine_specs.py

```diff
@@ -852,6 +852,20 @@ def _partition_query(
         """).format(**locals())
         return sql
 
+    @classmethod
+    def where_latest_partition(
+            cls, table_name, schema, database, qry, columns=None):
+        try:
+            col_name, value = cls.latest_partition(
+                table_name, schema, database, show_first=True)
+        except Exception:
+            # table is not partitioned
+            return False
+        for c in columns:
+            if c['name'] == col_name:
```

Comment:
NIT: `c.get()` is generally safer as it won't ever raise `KeyError`, though in this case the schema should be safe